tests/core/pyspec/eth2spec/test/helpers/multi_operations.py

```diff
@@ -257,3 +257,116 @@ def run_test_full_random_operations(spec, state, rng=Random(2080)):
 
     yield 'blocks', [signed_block]
     yield 'post', state
+
+
+def get_random_execution_requests(spec, state, rng):
+    deposits = get_random_deposits_requests(spec, state, rng)
+    withdrawals = get_random_withdrawals_requests(spec, state, rng)
+    consolidations = get_random_consolidations_requests(spec, state, rng)
+
+    execution_requests = spec.ExecutionRequests(
+        deposits=deposits,
+        withdrawals=withdrawals,
+        consolidations=consolidations
+    )
+
+    return execution_requests
+
+
+def get_random_deposits_requests(spec, state, rng, num_deposits=None):
+    if num_deposits is None:
+        num_deposits = rng.randrange(0, spec.MAX_DEPOSIT_REQUESTS_PER_PAYLOAD)
+
+    deposit_data_leaves = [spec.DepositData() for _ in range(len(state.validators))]
+
+    deposits_requests = []
+
+    for i in range(num_deposits):
+        index = rng.randrange(0, num_deposits)
+        withdrawal_pubkey = pubkeys[index]
+        withdrawal_credentials = spec.BLS_WITHDRAWAL_PREFIX + spec.hash(withdrawal_pubkey)[1:]
+
+        deposit, _, _ = build_deposit(
+            spec,
+            deposit_data_leaves,
+            pubkeys[index],
+            privkeys[index],
+            rng.randint(spec.MIN_ACTIVATION_BALANCE, spec.MAX_EFFECTIVE_BALANCE_ELECTRA),
+            withdrawal_credentials=withdrawal_credentials,
+            signed=True,
+        )
+
+        deposit_request = spec.DepositRequest(
+            pubkey=deposit.data.pubkey,
+            withdrawal_credentials=deposit.data.withdrawal_credentials,
+            amount=deposit.data.amount,
+            signature=deposit.data.signature,
+            index=deposit.data.index,
+        )
+        deposits_requests.append(deposit_request)
```

Comment:
For consistency with the others:
```suggestion
        )

        deposits_requests.append(deposit_request)
```